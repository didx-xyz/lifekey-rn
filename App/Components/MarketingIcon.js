// external dependencies
import React from "react"
import Svg, { Line, Polygon } from "react-native-svg"

const MarketingIcon = (props) => {
  const dynamic = {
    "stroke": props.stroke
  }

  return (
    <Svg width={props.width} height={props.height} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 23.4 21.4">
      <Line {...styles.cls1} {...dynamic} x1="22.9" y1="7.4" x2="0.5" y2="7.4"/>
      <Polygon {...styles.cls1} {...dynamic} points="11.7,20.9 0.5,7.4 5.5,0.5 17.9,0.5 22.9,7.4 11.7,20.9"/>
      <Polygon {...styles.cls1} {...dynamic} points="7.4,7.4 11.7,0.5 16,7.4 11.7,20.9 7.4,7.4"/>
      <Line {...styles.cls1} {...dynamic} x1="5.5" y1="0.5" x2="7.4" y2="7.4"/>
      <Line {...styles.cls1} {...dynamic} x1="17.9" y1="0.5" x2="16" y2="7.4"/>
    </Svg>
  )
}

MarketingIcon.defaultProps = {
  "stroke": "#000"
}

MarketingIcon.propTypes = {
  "width": React.PropTypes.number.isRequired,
  "height": React.PropTypes.number.isRequired,
  "stroke": React.PropTypes.string
}

const styles = {
  "cls1": {
    "fill": "none",
    "strokeLinecap": "round",
    "strokeLinejoin": "round"
  }
}

export default MarketingIcon
