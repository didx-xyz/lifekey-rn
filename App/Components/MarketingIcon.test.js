// external dependencies
import React from "react"

// internal dependencies
import MarketingIcon from "./MarketingIcon"

test("MarketingIcon renders correctly, by default", function() {
  const tree = shallow(
    <MarketingIcon width={16} height={16} />
  )

  expect(tree).toMatchSnapshot()
})

test("MarketingIcon renders correctly, when modified", function() {
  const tree = shallow(
    <MarketingIcon width={32} height={32} stroke="#fff" />
  )

  expect(tree).toMatchSnapshot()
})
