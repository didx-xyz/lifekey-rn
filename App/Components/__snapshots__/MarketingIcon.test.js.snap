// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketingIcon renders correctly, by default 1`] = `
<Svg
  height={16}
  preserveAspectRatio="xMidYMid meet"
  viewBox="0 0 23.4 21.4"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <Line
    fill="none"
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
    x1="22.9"
    x2="0.5"
    y1="7.4"
    y2="7.4"
  />
  <Polygon
    fill="none"
    points="11.7,20.9 0.5,7.4 5.5,0.5 17.9,0.5 22.9,7.4 11.7,20.9"
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
  <Polygon
    fill="none"
    points="7.4,7.4 11.7,0.5 16,7.4 11.7,20.9 7.4,7.4"
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
  <Line
    fill="none"
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
    x1="5.5"
    x2="7.4"
    y1="0.5"
    y2="7.4"
  />
  <Line
    fill="none"
    stroke="#000"
    strokeLinecap="round"
    strokeLinejoin="round"
    x1="17.9"
    x2="16"
    y1="0.5"
    y2="7.4"
  />
</Svg>
`;

exports[`MarketingIcon renders correctly, when modified 1`] = `
<Svg
  height={32}
  preserveAspectRatio="xMidYMid meet"
  viewBox="0 0 23.4 21.4"
  width={32}
  xmlns="http://www.w3.org/2000/svg"
>
  <Line
    fill="none"
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
    x1="22.9"
    x2="0.5"
    y1="7.4"
    y2="7.4"
  />
  <Polygon
    fill="none"
    points="11.7,20.9 0.5,7.4 5.5,0.5 17.9,0.5 22.9,7.4 11.7,20.9"
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
  <Polygon
    fill="none"
    points="7.4,7.4 11.7,0.5 16,7.4 11.7,20.9 7.4,7.4"
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
  />
  <Line
    fill="none"
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
    x1="5.5"
    x2="7.4"
    y1="0.5"
    y2="7.4"
  />
  <Line
    fill="none"
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
    x1="17.9"
    x2="16"
    y1="0.5"
    y2="7.4"
  />
</Svg>
`;
